engines/e_capi.c

```diff
@@ -888,8 +926,12 @@ int capi_rsa_sign(int dtype, const unsigned char *m, unsigned int m_len,
     /* Now cleanup */
 
  err:
+    if (hash)
     CryptDestroyHash(hash);
```

Comment:
If `CryptAcquireContext` fails on both tries, then `HCRYPTPROV hprov` will be unintialized and eventually released at `CryptReleaseContext`.

Double failure is not uncommon, especially if the container does not exist.